engine/runtime/src/main/java/org/enso/interpreter/runtime/data/ArrayOverBuffer.java

```diff
@@ -0,0 +1,43 @@
+package org.enso.interpreter.runtime.data;
+
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.interop.TruffleObject;
+import com.oracle.truffle.api.library.ExportLibrary;
+import com.oracle.truffle.api.library.ExportMessage;
+import org.enso.interpreter.dsl.Builtin;
+import org.enso.interpreter.runtime.library.dispatch.MethodDispatchLibrary;
+
+import java.nio.ByteBuffer;
+
+@ExportLibrary(InteropLibrary.class)
+public class ArrayOverBuffer implements TruffleObject {
+  private final ByteBuffer buffer;
+
+  private ArrayOverBuffer(ByteBuffer buffer) {
+    this.buffer = buffer;
+  }
+
+  @ExportMessage
+  Object readArrayElement(long index) {
+    return (long) buffer.get(Math.toIntExact(index));
+  }
+
+  @ExportMessage
+  boolean hasArrayElements() {
+    return true;
+  }
+
+  @ExportMessage
+  boolean isArrayElementReadable(long index) {
+    return index >= 0 && index < getArraySize();
+  }
+
+  @ExportMessage
+  long getArraySize() {
+    return buffer.remaining();
```

Comment:
`remaining` assumes the current `position=0`. Maybe you want to replace it with `limit`.

Reply:
Right! I assume that the buffer is in 'read mode' and the array should contain the content which would be 'read' from the buffer.

But indeed I forgot to offset the position, so I did the following update:
```diff
Object readArrayElement(long index) {
-    return (long) buffer.get(Math.toIntExact(index));
+    return (long) buffer.get(buffer.position() + Math.toIntExact(index));
}
```

Do you think that makes sense, or instead of indexing position...position+remaining, I should switch to 0...limit?